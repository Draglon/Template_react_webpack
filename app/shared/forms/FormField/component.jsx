import React from 'react';
import PropTypes from 'prop-types';
import { ErrorMessage } from 'formik';
import { Input } from 'antd';

const FormField = ({ field, field: { name }, form: { touched, errors }, fieldPrefix, ...rest }) => (
  <div className={`form__field${touched[name] && errors[name] ? ' form__field--error' : ''}`}>
    <Input {...field} {...rest} className="error" prefix={fieldPrefix} />
    <ErrorMessage name={name} component="div" />
  </div>
);

FormField.defaultProps = {
  fieldPrefix: null,
};

FormField.propTypes = {
  field: PropTypes.object.isRequired,
  form: PropTypes.object.isRequired,
  fieldPrefix: PropTypes.node,
};

export default FormField;
