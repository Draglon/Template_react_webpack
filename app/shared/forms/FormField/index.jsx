import FormField from './component';

export default FormField;
