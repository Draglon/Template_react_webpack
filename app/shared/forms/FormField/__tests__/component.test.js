import React from 'react';
import { Icon } from 'antd';

import FormField from '../component';

describe('<FormField /> component', () => {
  let props;
  let component;

  it('Render snapshot - with error', () => {
    props = {
      field: { name: 'name' },
      form: { touched: { name: true }, errors: { name: 'some error msg' } },
      fieldPrefix: false,
      type: 'text',
      placeholder: 'placeholder',
    };
    component = shallow(<FormField {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('Render snapshot - without error', () => {
    props = {
      field: { name: 'name' },
      form: { touched: { name: false }, errors: {} },
      fieldPrefix: false,
      type: 'text',
      placeholder: 'placeholder',
    };
    component = shallow(<FormField {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('Render snapshot - with field prefix and error', () => {
    props = {
      field: { name: 'name' },
      form: { touched: { name: true }, errors: { name: 'some error msg' } },
      fieldPrefix: <Icon type="lock" />,
      type: 'text',
      placeholder: 'placeholder',
    };
    component = shallow(<FormField {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('Render snapshot - with field prefix and without error', () => {
    props = {
      field: { name: 'name' },
      form: { touched: { name: false }, errors: {} },
      fieldPrefix: <Icon type="lock" />,
      type: 'text',
      placeholder: 'placeholder',
    };
    component = shallow(<FormField {...props} />);
    expect(component).toMatchSnapshot();
  });
});
