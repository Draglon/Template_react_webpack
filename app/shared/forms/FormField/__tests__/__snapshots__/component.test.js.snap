// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormField /> component Render snapshot - with error 1`] = `
<div
  className="form__field form__field--error"
>
  <Input
    className="error"
    name="name"
    placeholder="placeholder"
    prefix={false}
    type="text"
  />
  <FormikConnect(ErrorMessageImpl)
    component="div"
    name="name"
  />
</div>
`;

exports[`<FormField /> component Render snapshot - with field prefix and error 1`] = `
<div
  className="form__field form__field--error"
>
  <Input
    className="error"
    name="name"
    placeholder="placeholder"
    prefix={
      <Icon
        type="lock"
      />
    }
    type="text"
  />
  <FormikConnect(ErrorMessageImpl)
    component="div"
    name="name"
  />
</div>
`;

exports[`<FormField /> component Render snapshot - with field prefix and without error 1`] = `
<div
  className="form__field"
>
  <Input
    className="error"
    name="name"
    placeholder="placeholder"
    prefix={
      <Icon
        type="lock"
      />
    }
    type="text"
  />
  <FormikConnect(ErrorMessageImpl)
    component="div"
    name="name"
  />
</div>
`;

exports[`<FormField /> component Render snapshot - without error 1`] = `
<div
  className="form__field"
>
  <Input
    className="error"
    name="name"
    placeholder="placeholder"
    prefix={false}
    type="text"
  />
  <FormikConnect(ErrorMessageImpl)
    component="div"
    name="name"
  />
</div>
`;
